xivo_bus/mixins.py

```diff
@@ -376,32 +376,47 @@ def __init__(self, service_uuid=None, **kwargs):
         super(WazoEventMixin, self).__init__(**kwargs)
         self.service_uuid = service_uuid
 
-    def __serialize_event(self, event):
+    def __generate_payload(self, event, headers, initial_data):
+        payload = {}
+        data = initial_data.copy() if initial_data else {}
         try:
-            return event.marshal()
+            data.update(event.marshal())
         except AttributeError:
             self.log.exception("Received invalid event '%s'", event)
             raise ValueError('Not a valid Wazo Event')
+        else:
+            payload.update(headers, data=data)
+            return payload
+
+    def __generate_headers(self, event, extra_headers):
+        headers = {}
+        headers.update(extra_headers or {})
 
-    def __generate_metadata(self, event):
-        metadata = {
-            'name': event.name,
-            'origin_uuid': self.service_uuid,
-            'timestamp': datetime.now().isoformat(),
-        }
+        try:
+            headers.update(event.headers)
+        except AttributeError:
+            pass
+
+        if hasattr(event, 'required_access'):
+            headers['required_access'] = event.required_access
+
+        # TODO: remove deprecated `required_acl`
```

Comment:
Should this be removed now? or is it for later?

Reply:
This will be removed at a future time, to allow a transition time to avoid creating breakage